packages/config-service/src/services/validationService.ts

```diff
@@ -11,13 +11,21 @@ export class ValidationService {
     // validate mandatory fields and their types
     Object.entries(GlobalConfig.ENTRIES).forEach(([entryName, entryInfo]) => {
       if (entryInfo.required) {
-        if (!envs.hasOwnProperty(entryName)) {
+        if (!Object.prototype.hasOwnProperty.call(envs, entryName)) {
           throw new Error(`Configuration error: ${entryName} is a mandatory configuration for relay operation.`);
         }
 
         if (entryInfo.type === 'number' && isNaN(Number(envs[entryName]))) {
           throw new Error(`Configuration error: ${entryName} must be a valid number.`);
         }
+
+        if (entryInfo.type === 'array' && envs[entryName]) {
+          try {
+            JSON.parse(envs[entryName] as string);
```

Comment:
we are doing this `JSON.parse` twice, right?

Reply:
Yes, once in startUp for validation and once in typeCasting for the actual conversion